build-aux/pkg.nix

```diff
@@ -118,8 +118,6 @@ in stdenv.mkDerivation (finalAttrs: {
 
   postPatch = ''
     patchShebangs build-aux/*.sh build-aux/git-version-gen
-  '' + lib.optionalString stdenv.isDarwin ''
-    sed -i -e 's|@import AppKit;|#import <AppKit/AppKit.h>|' src/macfonts.m
   '';
 
   NIX_LDFLAGS = lib.optionalString stdenv.isDarwin "-framework AppKit";
```

Comment:
Could it be this can be removed as well?

Reply:
I don't know. Nor do I know how to trigger a Darwin build to find out.

It seems like if this had to get added at all, nothing has changed that would make it irrelevant now. The sed expression removal works because we just fixed it to be that way upstream. On the other hand Homebrew doesn't seem to need this LDFLAG (or perhaps it has it by default?).

Reply:
> I don't know. Nor do I know how to trigger a Darwin build to find out.
> 
> It seems like if this had to get added at all, nothing has changed that would make it irrelevant now. The sed expression removal works because we just fixed it to be that way upstream. On the other hand Homebrew doesn't seem to need this LDFLAG (or perhaps it has it by default?).

Hmm I wonder whether this is still relevant. According to `git blame` those additions were done long ago:

https://github.com/nixos/nixpkgs/commit/4a8b2c29420ef11af6d9abad8d8b6710eb397a23

In general, if you want to setup a Darwin & Nix CI, I recently heard of this interesting project:

https://github.com/MatthewCroughan/NixThePlanet/#macos 

You can start by testing the build on that VM by yourself. Plugging that machine into a GitHub CI would be a nice Nix miniproject, but let's leave that aside for now.

Another option would be to use Nixpkgs' CI to test that - i.e open a Nixpkgs draft PR, and see how the Darwin builds get along. This also takes time though, as our CI runner has many Darwin requests to process.

You could also commit the Linux related removals, and only deal with these `NIX_LDFLAGS` later.

Reply:
Thank for the options. I'll look into them, some now some later.

For the moment I think we'll move ahead with the bits I'm sure about, then when v0.15.0 actually drops and we update the official Nix pkgs repo with the stable release we can test whether than is needed or not. If not we can then remove it from the Flake too.